django/template/backends/django.py

```diff
@@ -25,6 +37,54 @@ def __init__(self, params):
         super().__init__(params)
         self.engine = Engine(self.dirs, self.app_dirs, **options)
 
+    def check(self, **kwargs):
+        errors = []
+        errors.extend(self._check_string_if_invalid_is_string())
+        errors.extend(self._check_for_template_tags_with_the_same_name())
+        return errors
+
+    @staticmethod
+    def _check_string_if_invalid_is_string():
```

Comment:
Okay, you’ve missed part of the motivation behind the ticket. The `check()` method should check *this backend only*. It should be an instance method, not a static one:

```suggestion
    def _check_string_if_invalid_is_string(self):
```

Then the method should check the configuration of only *this* backend via e.g. `self.engine`. Looping over `settings.TEMPLATES` checks *all* backends.

Both check methods need changing this way.

Reply:
Ah that's right, leaving the `settings.TEMPLATES` in would mean checking all engines, for all engines. So, updating `_check_string_if_invalid_is_string` to behave as an instance method should be straightforward. The `_check_for_template_tags_with_the_same_name` is an interesting case though, because as far as I can tell, the logic is not `Engine` instance specific. It needs to check for name duplicates across backends so it probably requires some kind of access to `settings.TEMPLATES` (?). The way I updated this check, it runs only  for `DjangoTemplates` but it also may do duplicate work for the >1 `DjangoTemplates` backends case, which is not great. The other alternative is to leave that as a _global_ check in `checks/templates.py` but I don't like this much either since it kind of goes against our plan here.

Reply:
It’s fine to repeat `_check_for_template_tags_with_the_same_name` per backend. It still depends on the per-backend `libraries` option, so one backend may have a clash whilst another doesn’t.

Reply:
> But there’s no problem with two backends using the same name for different libraries.

I see. This confused me a bit because there's an explicit test for the 2 DjangoTemplates backends case, which expects to get the warning: https://github.com/django/django/blob/b07e2d57a000d98c73492e5242fed91d502a780a/tests/check_framework/test_templates.py#L145
So probably the test is wrong/needs to be updated?
(I should have mentioned this test already, my bad)

Reply:
Giving this another thought it's obvious that the test needs updating, so I'm pushing changes to match that behavior. Thank you for the guidance Adam.